core/src/test/java/org/apache/calcite/test/SqlHintsConverterTest.java

```diff
@@ -771,14 +788,42 @@ static HintStrategyTable createHintStrategies(HintStrategyTable.Builder builder)
                         + "allowed options: [ONE_PHASE, TWO_PHASE]",
                     hint.hintName)).build())
         .hintStrategy("use_hash_join",
-          HintPredicates.and(HintPredicates.JOIN, joinWithFixedTableName()))
+          HintPredicates.or(
+              HintPredicates.and(HintPredicates.CORRELATE, temporalJoinWithFixedTableName()),
+              HintPredicates.and(HintPredicates.JOIN, joinWithFixedTableName())))
         .hintStrategy("use_merge_join",
             HintStrategy.builder(
                 HintPredicates.and(HintPredicates.JOIN, joinWithFixedTableName()))
                 .excludedRules(EnumerableRules.ENUMERABLE_JOIN_RULE).build())
         .build();
     }
 
+    /** Returns a {@link HintPredicate} for temporal join with specified table references. */
+    private static HintPredicate temporalJoinWithFixedTableName() {
+      return (hint, rel) -> {
+        if (!(rel instanceof LogicalCorrelate)) {
+          return false;
+        }
+        LogicalCorrelate correlate = (LogicalCorrelate) rel;
+        RelNode leftInput = correlate.getLeft();
+        if (!(leftInput instanceof TableScan)) {
+          return false;
+        }
+        RelNode rightInput = correlate.getRight();
+        boolean rightIsFilterOnSnapshot = rightInput instanceof Filter
+            && ((Filter) rightInput).getInput() instanceof Snapshot
+            && ((Snapshot) ((Filter) rightInput).getInput()).getInput() instanceof TableScan;
+        if (!rightIsFilterOnSnapshot) {
+          return false;
+        }
+        String leftTableName = getTableNameOfScan(leftInput);
+        String rightTableName = getTableNameOfScan(
+            ((Snapshot) ((Filter) rightInput).getInput()).getInput());
+        final List<String> tableNames = hint.listOptions;
+        return tableNames.contains(leftTableName) && tableNames.contains(rightTableName);
+      };
```

Comment:
Can we use `equalsStringList` here ?

Reply:
> And there must have a filter ?
Yes for temporal join. Because there must a filter which contain join condition.

> Can we use equalsStringList here ?
I'm not sure if there is necessary here to build a list which contain leftTableName and rightTableName, then call equalsStringList. Because we have already get  leftTableName and rightTableName separately.